v3/src/models/data/data-set.ts

```diff
@@ -791,6 +801,17 @@ export const DataSet = types.model("DataSet", {
         }
       },
 
+      applyAttributeProperties(attributeID: string, attrProps: IAttributeSnapshot) {
+        (attrProps.name != null) && this.setAttributeName(attributeID, attrProps.name)
+        const attribute = attributeID && attrIDMap[attributeID]
+        if (!attribute) return
+        ;(attrProps.description != null) && attribute.setDescription(attrProps.description)
+        ;(attrProps.userType != null) && attribute.setUserType(attrProps.userType)
+        ;(attrProps.units != null) && attribute.setUnits(attrProps.units)
+        ;(attrProps.precision != null) && attribute.setPrecision(attrProps.precision)
+        ;(attrProps.editable != null) && attribute.setEditable(attrProps.editable)
```

Comment:
I thought about that, but I wasn't sure it would handle the special cases (like `undefined`) properly. I'm also not sure that the code above is necessarily handling it properly either, but I'm inclined to leave it alone for now.